Kernel/Bus/USB/UHCI/UHCIController.cpp

```diff
@@ -556,6 +600,31 @@ ErrorOr<void> UHCIController::spawn_port_process()
     return {};
 }
 
+ErrorOr<void> UHCIController::spawn_async_poll_process()
+{
+    LockRefPtr<Thread> async_poll_thread;
+    (void)Process::create_kernel_process(async_poll_thread, TRY(KString::try_create("UHCI Async Poll Task"sv)), [&] {
+        u16 poll_interval_ms = 1024;
+        for (;;) {
+            SpinlockLocker locker(m_async_lock);
+
+            for (auto& handle : m_active_async_transfers) {
+                poll_interval_ms = min(poll_interval_ms, handle.ms_poll_interval);
```

Comment:
micro-optimization, but you could could consider pre-calculating the min poll interval when appending to the queue and avoid having to re-calculate it every time around the loop.

e.g.

```
void add_transfer(Transfer* t, u16 poll_interval_ms)
{
    // setup some stuff
    
    Locker lock(m_lock);
    if (m_queue->min_poll_interval_ms > poll_interval_ms)
       m_queue->min_poll_interval_ms = poll_interval_ms;
    m_queue->add(t);
}
```
And then re-calculate the min after removing the item or items that have the min poll interval.